include/ideep/operators/conv.hpp

```diff
@@ -905,52 +1346,62 @@ struct convolution_forward
     auto oc = groups * dims_in[0 + grouped];
     if (5 == src_size) {
       kernel_size.push_back(dims_in[ndims - 3]);
+      kernel_size.push_back(dims_in[ndims - 2]);
+    } else if (4 == src_size) {
+      kernel_size.push_back(dims_in[ndims - 2]);
     }
-    kernel_size.push_back(dims_in[ndims - 2]);
     kernel_size.push_back(dims_in[ndims - 1]);
     if (src_dims.empty()) {
-      // Construct a dummy case
-      x_dims.push_back(1);
+      // Construct a dummy case. Shape from resnet50 model.
+      x_dims.push_back(32);
       x_dims.push_back(ic);
-      y_dims.push_back(1);
+      y_dims.push_back(32);
       y_dims.push_back(oc);
+      x_dims.push_back(14 * kernel_size[0]);
       if (4 == src_size) {
-        x_dims.push_back(4 * kernel_size[0]);
-        x_dims.push_back(4 * kernel_size[1]);
-      } else {
-        x_dims.push_back(8 * kernel_size[0]);
-        x_dims.push_back(8 * kernel_size[1]);
-        x_dims.push_back(8 * kernel_size[2]);
+        x_dims.push_back(14 * kernel_size[1]);
+      } else if (5 == src_size) {
+        x_dims.push_back(14 * kernel_size[1]);
+        x_dims.push_back(14 * kernel_size[2]);
       }
```

Comment:
A note here: Potential perf impact.

Reply:
The shape is used to get preferred weight descriptor. Do you mean that the weight desc may be improper with such an input shape?

Reply:
No, I wanted to put a caveat that this change might cause perf diff in the stock pytorch. Not something we should address, just a note.

Reply:
OK